### PR TITLE
fix(aws-lambda-dynamodb) Added VPC support for the aws-lambda-dynamodb construct.

### PR DESCRIPTION
**Pull request by danielmatuki**
*Issue #, if available:*
https://github.com/awslabs/aws-solutions-constructs/issues/148

*Description of changes:*
Implemented changes to will allow the Lambda Functions deployed by this pattern to access resources in a VPC.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: c1ea3aca3a07d5ddcd1eb64bbc5372c2751b3a0d
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8fe899b8-8493-4d08-a25c-2525e5f75e7e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: d4932e9bc62858ea4b3a7bcd6f8e54fe9c61509a
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8bf7ca8a-2672-4125-84f4-209c622e1e10%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by danielmatuki** [comment]
@biffgaut @hnishar Could you please review this PR?

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 5d01d84b90766d597a060bd8b0f85ff61db8a74d
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d3edeb16-1735-4bbe-9b90-a599fda43270%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/test/integ.deployFunctionWithExistingVpc.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/test/integ.deployFunctionWithExistingVpc.ts:1]: Nice touch adding this test - looks like only the Sagemaker construct included this test up to this point.

**Review by biffgaut — COMMENTED** [review]
Looks good. Before we merge it, did you test it in a client app using these procedures?

https://w.amazon.com/bin/view/AWS/Teams/SA/AWS_Solutions_Builder/Solutions_Constructs_Home/Testing_Constructs_From_Repos/

File: source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/test/integ.deployFunctionWithExistingVpc.ts (hunk 1 above)
Comment (danielmatuki) [reply source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/test/integ.deployFunctionWithExistingVpc.ts:1]: Hi @biffgaut I've created the sample application using the construct with a VPC and validated that the Lambda function was able to write to the DDB table.

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 9a8d5956f1eb66313050c7687560f9ee6e0b07d6
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2712b311-e6a2-4056-a222-1345d5fec2a4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: a7686e8a857e201aa660ab9ef5d56b57e1bfa03f
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=583c213c-cd67-4004-9844-f65449ad6371%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by danielmatuki** [comment]
Hi @biffgaut I've created the sample application using the instructions provided and deployed the construct with a VPC and validated that the Lambda function was able to write to the DDB table. Please let me know if this looks good, thanks,

**Review by biffgaut — APPROVED** [review]

**Pull request merged**